modules/custom/activity_basics/src/Plugin/ActivityEntityCondition/NodePublishedActivityEntityCondition.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drupal\activity_basics\Plugin\ActivityEntityCondition;
+
+use Drupal\activity_creator\Plugin\ActivityEntityConditionBase;
+
+/**
+ * Provides an activity condition to check whether a node is published or not.
+ *
+ * @ActivityEntityCondition(
+ *  id = "node_published",
+ *  label = @Translation("Entity Published"),
+ *  entities = {"node" = {}}
+ * )
+ */
+class NodePublishedActivityEntityCondition extends ActivityEntityConditionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isValidEntityCondition($entity) {
```

Comment:
Did you check what will happen with entities like Post? You are adding it now to `message.template.create_content_in_joined_group.yml` which is also triggered for posts, did you check if this doesn't apply there? 
Returning FALSE might not cut it.